rfd/0056-cockroach-backend.md

```diff
@@ -0,0 +1,70 @@
+---
+authors: Jim Bishopp (jim@goteleport.com)
+state: draft
+---
+
+# RFD 56 - CockroachDB Backend
+
+## What
+
+A CockroachDB backend extends Teleport with the ability to store data
+in a self-hosted CockroachDB cluster.
+
+## Why
+
+Supporting additional backends such as CockroachDB reduces onboarding time
+for Teleport customers by allowing them to use existing infrastructure.
+
+## Scope
+
+This RFD focuses on implementing a backend for a self-hosted CockroachDB cluster.
+
+Requirements:
+
+- Prioritize self-hosted CockroachDB.
+- Implementation should support CockroachDB node failures by automatically
+  reconnecting to another node when a connected node becomes unavailable.
+- Design should consider extensibility to support other SQL databases 
+  such as PostgreSQL and MySQL.
+- Design should utilize a dialer/connector abstraction when connecting to
+  the database to allow the implementation to be extended to support SQL
+  databases hosted by a cloud provider.
+
+## Authentication
```

Comment:
Add a section about cloud authentication too? I.e. using IAM auth for RDS, Cloud SQL, etc.

Reply:
Will do. I wasn't focused on cloud initially.